pallets/gas/src/lib.rs

```diff
@@ -178,15 +180,15 @@ pub mod pallet {
                             // this is specified, so it need to get to the first specified parent also
                             // going up until external or specified parent is found
 
-                            let (parent_key, mut parent_node) = Self::node_with_value(parent);
-
+                            let (parent_key, mut parent_node) = Self::node_with_value(parent)
+                                .expect("Upstream node with value always exists for this key");
```

Comment:
same here. aren't we able to avoid that expect?

Reply:
This is a basic invariant: for every node in the tree there is a node with value (which can be the node itself if it has type other than `ValueType::UnpecifiedLocal`). If this invariant doesn't hold,  the validator node must be restarted due to storage corruption. I don't see a way for us to gently handle such situation rather than panicking.

Reply:
Some guarantees is ok, maybe just add a comment why this expect will never fire @ekovalev